### ISSUE TITLE
Support additionnal commands when deploying review apps

### ISSUE THREAD

**Issue by nicolas-besnard** [dokku/ci-docker-image#9]
I would like to know if there's a way to extend this to allow additionnal commands to run when deploying review apps?

For example, I'd like to be able to set different ENV variables for my app (database url, different 3rd party credentials etc...).

Being able to run a script would be a plus (create / remove a database created for the review app)

**Comment by jbosse** [comment]
I had a similar need where I wanted to set my current tag to `VERSION` in my app. 
I used to deploy manually with the run action and I have cobbled this together, you might find it helpful:
```yaml
      - name: Update VERSION to new_tag
        run: |
          mkdir -p "$HOME/.ssh"
          touch "$HOME/.ssh/known_hosts"
          echo "${{ secrets.SSH_PRIVATE_KEY }}" > "$HOME/.ssh/deploy_key"
          chmod 700 "$HOME/.ssh"
          chmod 600 "$HOME/.ssh/known_hosts"
          chmod 600 "$HOME/.ssh/deploy_key"
          eval $(ssh-agent)
          ssh-add "$HOME/.ssh/deploy_key"
          ssh -o "StrictHostKeyChecking=no" ssh://dokku@${{ secrets.DOKKU_HOST }}:${{ secrets.DOKKU_PORT }} config:set --no-restart {{ secrets.DOKKU_APP_NAME }} VERSION=${{steps.tag_version_dry_run.outputs.new_tag}}  
```

**Comment by clemens** [comment]
Here's a GitLab example that works just fine:

``` yaml
review_app:
  # other stuff like rules
  script:
    dokku-deploy
    ssh_remote="ssh://dokku@$(parse-ssh-host):$(parse-ssh-port)"

    # then do anything you need to do with this ssh_remote
    ssh "$ssh_remote" -- postgres:create some-other-database
    ssh "$ssh_remote" -- postgres:link some-other-database your-app-name
```

The only downside of this approach is that almost every command causes a re-deploy of your app, so it's not terribly efficient. I think you could go with custom plugins if you want to avoid that.

**Comment by josegonzalez** [comment]
Going to close this. I don't really think this sort of logic goes into this image. For anyone looking for better CI/CD experience, users are better suited to running Ansible from CI instead of using this image to run any generic commands through it (the ansible commands cover a wide range of features and provide better idempotency than generic commands might).

**Comment by josegonzalez** [comment]
I implemented this in #28 as a file hook. I think that strikes the right balance between functionality and complexity of the implementation. Hopefully this is useful to you and others implementing similar pre-deploy configuration.
